pragma-node/src/utils/signing/mod.rs

```diff
@@ -30,3 +40,31 @@ pub fn sign_data(signer: &SigningKey, data: &impl Signable) -> Result<String, Si
         .map_err(SigningError::SigningError)?;
     Ok(format!("0x{:}", signature))
 }
+
+/// Assert that a signature (passed with the request for creating new
+/// entries) is correctly signed by the publisher and in a valid format.
+/// NOTE: the [is_legacy] parameter can be set to True for signatures
+/// that uses our SDK before version 2.0.
+pub fn assert_signature_is_valid<T>(
```

Comment:
I'd prefer to have an `assert_legacy_signature_is_valid` separate function instead of the boolean flag

Reply:
done